### PR TITLE
michiel/change client side flow registration push notification

### PR DESCRIPTION
**Pull request by michielmulders**
Change flow client-side using upsert logic from https://github.com/luniehq/lunie-api/pull/678 for push registration.

### CODE REVIEW

File: src/vuex/modules/pushNotifications.js [hunk 1]
```diff
@@ -66,25 +66,41 @@ const askPermissionAndRegister = async (activeNetworks, apollo) => {
     } catch (error) {
       console.log("Permission denied", error)
     }
+  } else {
+    // upsert
+    const token = localStorage.getItem("registration-push-notifications-token")
+    registerDevice(token, activeNetworks, apollo, true) // non-blocking
```
Comment (Bitcoinera) [inline src/vuex/modules/pushNotifications.js:72]: So the logic is that every time a registered device logs in, the token does in fact change, or why do we need to update these infos?

Comment (michielmulders) [reply src/vuex/modules/pushNotifications.js]: I've updated the flow a bit as this was not correct.

There are three options now:
1. device not registered === null
2. blocked push notifications === blocked
3. allowed push notifications === allowed

if null -> ask permission
if block -> skip flow
if allowed -> update device registration async when logging in (to detect new addresses so we can also create default topic subscriptions for those addresses/network).

File: src/vuex/modules/pushNotifications.js [hunk 2]
```diff
@@ -57,22 +67,27 @@ const askPermissionAndRegister = async (activeNetworks, apollo) => {
...
           await registerDevice(token, activeNetworks, apollo)
         })
-        .catch(error =>
-          console.log(
-            "bug FCM throws error while deleting token on first refresh",
-            error
-          )
-        )
+        .catch(() => {
+          localStorage.setItem("registration-push-notifications", "blocked")
```
Comment (faboweb) [inline src/vuex/modules/pushNotifications.js:71]: is the only error that can happen, that the user blocked the notification?

Comment (Bitcoinera) [reply src/vuex/modules/pushNotifications.js:71]: I made the exact same comment on Slack! 🤣  amazing

Comment (Bitcoinera) [reply src/vuex/modules/pushNotifications.js:71]: neyo-1oC7B5v1VzaxJC9KeQ

Comment (faboweb) [reply src/vuex/modules/pushNotifications.js:71]: hahahah we already think the same! dangerous :D

Comment (Bitcoinera) [reply src/vuex/modules/pushNotifications.js:71]: xD

Comment (michielmulders) [reply src/vuex/modules/pushNotifications.js:72]: Sorry to interrupt this ET party 😄 

After some deep deep googling I found that you can check for the code:
1. error.code = blocked => set localstorage
2. other errors? console.error

Good catch to improve error catching, thanks guys! 👍

File: src/vuex/modules/pushNotifications.js [hunk 3]
```diff
@@ -28,51 +28,67 @@ const initializeFirebase = async () => {
...
+
+        // Granted? Store device
+        const token = await messaging.getToken()
+        await registerDevice(token, activeNetworks, apollo)
+      })
+      .catch(error => {
+        if (error.code === "messaging/permission-blocked") {
+          localStorage.setItem("registration-push-notifications", "blocked")
+        } else {
+          console.error("Error occurred during permission request for push notifications", error);
```
Comment (faboweb) [inline src/vuex/modules/pushNotifications.js:74]: add Sentry here so we get notified about issues

**Pull request merged**